src/app/dashboard/projects/list-project/project-item/project-item.component.ts

```diff
@@ -207,10 +207,12 @@ export class ProjectItemComponent implements OnInit, OnDestroy {
 
   showSettings() {
     if (this.status.netStatus) {
+      // HACK: make a clone of project because we change it settings and dispatch it.
+      const project = cloneDeep(this.project);
       this.modalService.show({
         component: ProjectSettingsModalComponent,
         inputs: {
-          project: this.project,
+          project,
```

Comment:
I think it is better to pass only the property we need not all the project's props

Reply:
Hmmm, this is the project setting modal that has info and actions related to all of the project props
Also we need to do a fully refactor code some day 🙂

Reply:
and I see we clone the project again inside [project-details.component.ts](https://github.com/Swing-team/pendulums-web-client/blob/dad8a362699640120b6c217d6c278414fa04515d/src/app/dashboard/projects/settings/details/project-details.component.ts#L42)

am I right, or there is a reason we are doing this?

Reply:
There is a reason for that
After the modal is initialized, if we do some changes to project probs and dispatch them into store, the input project won't change. So we need to change it to show changes in real time like remove a team member and etc. 
New versions of ngrx makes writable of subscribed object false. So to change the props we needed to clone project.
There are some configs that makes the state writable but it is better to set to false because we can see our performance issues and make them better.